Networking/Networking/Mapper/SitePluginMapper.swift

```diff
@@ -7,7 +7,14 @@ struct SitePluginMapper: Mapper {
     /// Site Identifier associated to the plugin that will be parsed.
     /// We're injecting this field via `JSONDecoder.userInfo` because the remote endpoints don't return the SiteID in the plugin endpoint.
     ///
-    let siteID: Int64
+    private let siteID: Int64
+
+    private let withDataEnvelope: Bool
+
+    init(siteID: Int64 = -1, withDataEnvelope: Bool = true) {
```

Comment:
I'm not very familiar with cookie authentication, how can we fetch site plugins without a site ID - like with the site URL instead? Is it because these sites don't have a site ID since they're self-hosted?

In any case, how about a nullable `siteID` instead of a `-1` default value? I'm just thinking that if the site ID update is missed somewhere by accident then this invalid site ID is used throughout the app which results in an unexpected app state.

Maybe the site identifier can become an enum with an associated identifier value, like one case with `siteID: Int64` and the other with `siteURL: String` (whatever identifies the site without a site ID).

Reply:
In this PR, we're managing plugins directly with [WordPress.org REST API endpoints](https://developer.wordpress.org/rest-api/reference/plugins/), meaning we are communicating directly with the site and not through Jetpack-tunneled requests, so site ID is not necessary. We don't have a site ID because the site isn't associated to the user's WP.com account yet (not until after the Jetpack connection step completes).

During login (unauthenticated state), we don't intend to save the plugin details to the storage, so having any form of identification for the site is unnecessary.

I've updated the comment for `SitePluginMapper` initializer in f47e0e080766d96ab742c4e48e068cc9416e3cbf, I hope it's helpful.